examples/bzlmod/BUILD.bazel

```diff
@@ -21,6 +21,11 @@ compile_pip_requirements_3_9(
     requirements_windows = "requirements_windows_3_9.txt",
 )
 
+uv_pip_compile(
```

Comment:
Yeah, it's weird, but thats what the actual CLI is invoked as: `uv pip compile`. They bolted on a bunch of `pip` subcommands namespaced under `pip`.

We can discuss more later, but I have some ideas. e.g. Just because we have a `uv` toolchain, it doesn't mean all our rules need to be called `uv_foo`. Sure, we can have `uv_pip_compile` to support / replace the existing `compile_pip_requirements`.

But I hope when the overall feature lands, that we can aim for more meaningful extensions or targets added to the repositories as well?

e.g.
```
pypi = use_extension("@rules_python//pypi:extension.bzl", "pypi")
pypi.from_lockfile(
    name = "pypi",
    lockfile = "//third_party:requirements.txt",
    dependencies = ["//third_party:pyproject.toml"],
)
use_repo(pypi, "pypi")
```

So internally, if we use a `uv` toolchain, it isn't exposed to the user as the primary interface. Sure, they can grab it as a toolchain and do what they like if they are power users, but otherwise, but otherwise, we only need to support a basic capability to produce and consume a lockfile from a standard pyproject.toml